tests/jax/components/executing/action_selection.py

```diff
@@ -0,0 +1,237 @@
+            "networks": {
+                agent_net_keys["agent_0"]: SimpleNamespace(get_action=get_action),
+                agent_net_keys["agent_1"]: SimpleNamespace(get_action=get_action),
+                agent_net_keys["agent_2"]: SimpleNamespace(get_action=get_action),
+            }
+        }
+        key = jax.random.PRNGKey(5)
+        action_info = "action_info_test"
+        policy_info = "policy_info_test"
+
+        store = SimpleNamespace(
+            is_evaluator=None,
+            observations=observations,
+            observation=SimpleNamespace(observation=[0, 0, 1, 0], legal_actions=3),
+            agent="agent_0",
+            networks=networks,
+            agent_net_keys=agent_net_keys,
+            key=key,
+            action_info=action_info,
+            policy_info=policy_info,
+        )
+        self.store = store
+
+    def select_action(
+        self, agent: str, observation: NestedArray, state: NestedArray = None
+    ) -> NestedArray:
+        action_info = "action_info_" + str(agent) + "_" + str(observation)
+        policy_info = "policy_info_" + str(agent) + "_" + str(observation)
+        return action_info, policy_info
+
+
+@pytest.fixture
+def mock_executor() -> Executor:
+    """Mock executore component."""
+    return MockExecutor()
+
+
+# Test initiator
+def test_constructor(dummy_config: type) -> None:
+    """Test adding config as an attribute
+
+    Args:
+        dummy_config
+    """
+    ff_executor_select_action = FeedforwardExecutorSelectAction(config=dummy_config)
+    assert ff_executor_select_action.config.parm_0 == dummy_config.parm_0
+
+
+def test_constructor_empty_config(dummy_empty_config: type) -> None:
+    """Test adding empty config as an attribute
+
+    Args:
+        dummy_empty_config: ExecutorSelectActionConfig()
+    """
+    ff_executor_select_action = FeedforwardExecutorSelectAction(
+        config=dummy_empty_config
+    )
+    assert (
+        ff_executor_select_action.config.__dict__
+        == ExecutorSelectActionConfig().__dict__
+    )
+
+
+# Test on_execution_select_actions
+def test_on_execution_select_actions_with_empty_observations(
+    mock_empty_executor: Executor, dummy_config: type
+) -> None:
+    """Test on_execution_select_actions with empty observations
+
+    Args:
+        mock_empty_executor: executor with no observations and no agents
+    """
+    ff_executor_select_action = FeedforwardExecutorSelectAction(dummy_config)
+    ff_executor_select_action.on_execution_select_actions(executor=mock_empty_executor)
+
+    assert mock_empty_executor.store.actions_info == {}
+    assert mock_empty_executor.store.policies_info == {}
+
+
+def test_on_execution_select_actions(
+    mock_executor: Executor, dummy_config: type
+) -> None:
+    """Test on_execution_select_actions.
+
+    Args:
+        dummy_config: config
+        mock_executor: Executor
+    """
+
+    ff_executor_select_action = FeedforwardExecutorSelectAction(dummy_config)
```

Comment:
This can be made into a fixture. The exact same line is used elsewhere in the code